### DISCUSSION TITLE
Canvas children disappeared when using native aot in linux

### DISCUSSION THREAD

**Discussion by charleypeng** [AvaloniaUI/Avalonia#12406] [Q&A] [answered]
I use Canvas to draw some test controls and it worked fine when debugging, but when I publish as an Aot native, the aot  binary can be opened but the canvas children are gone(both windows and linux). 

p.s.  The racing chart which  using  livechartscore   seems to be laggy  in the native aot mode.

here is the  [releases](https://github.com/charleypeng/Cagaya_Radar/releases/tag/v0.3)  I've built in two versions

Please make sure that Font Noto and it's extras are installed; 

- Avalonia version: 11.0.0.1
- OS: ubuntu22.04
- Desktop Environment: kde

**Comment by timunie** [comment]
You need to make sure your dependencies are not trimmed. Avalonia is trimming friednly, but if you or any lib uses reflection, you need to handle it somehow. Not an Avalonia issue imo.

**Comment by charleypeng** [comment, marked as answer]
> 

thanks!  I have solved this problem by just using the dotnet publish command line instead of Rider. I think there may be something I set gone wrong in Rider.   

But as I mentioned the "p.s. The racing chart which using livechartscore seems to be laggy in the native aot mode"  I have no clue about this :(

**Comment by timunie** [comment]
livechartscore sounds like a 3rd party lib. So not and Avalonia issue as well. Let me change this into a Discussion.
